src/user/wikis/Wikis.js

```diff
@@ -0,0 +1,243 @@
+  constructor(props) {
+    super(props);
+    this.state = {
+      wikis: true,
+      allWikis: [],
+      currentPage: 1,
+      editorMode: false,
+      historyMode: false,
+      viewHistory: false,
+      newPageEditor: false,
+      sidebarEditor: false,
+      spinner: "Loading...",
+    };
+    this.axiosCancel = axios.CancelToken.source();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({
+      allWikis: nextProps.wikis.wikis,
+      spinner: "",
+    });
+  }
+
+  componentDidMount() {
+    const { getWikis } = this.props
+    setTimeout(() => {
+      getWikis();
+    });
+  }
+
+  handleEditorMode = (newPage = false, sidebar = false) => {
+    this.setState({
+      editorMode: true,
+      newPageEditor: !!newPage,
+      sidebarEditor: !!sidebar,
+    });
+  };
+
+  cancelEditor = () => {
+    this.setState({
+      editorMode: false,
+      newPageEditor: false,
+    });
+  };
+
+  setView = async (page, ref) => {
+    let pos = 0;
+    const { allWikis } = this.state
+    allWikis.forEach((ele, index) => {
+      if (ele.title === page) pos = index;
+    });
+    if (pos) {
+      this.setState(
+        {
+          spinner: "Switing Page.... ",
+        },
+        fetchPage.bind(this, page, pos)
+      );
+    } else {
+      this.handleEditorMode(true);
+    }
+  };
+
+  handleSave = async (page, newPage = false, sidebar = false) => {
+    const { allWikis } = this.state;
+    if (!newPage && !sidebar) {
+      this.setState(
+        {
+          spinner: "Saving... ",
+        },
+        saveChangesToPage.bind(this, page, newPage, sidebar)
+      );
+    } else if (sidebar) {
+      this.setState(
+        {
+          spinner: "Saving...",
+        },
+        saveChangesToPage.bind(this, page, newPage, sidebar)
+      );
+    } else {
+      if (allWikis.filter((ele) => ele.title === page.title).length === 0) {
+        this.setState(
+          {
+            spinner: "Creating New Page...",
+          },
+          saveChangesToPage.bind(this, page, newPage, sidebar)
+        );
+      } else {
+        toast.error("Page with that title already exsits!");
+      }
+    }
+  };
+
+  deletePage = async () => {
+    const { allWikis, currentPage } = this.state;
+    this.setState(
+      {
+        spinner: "Deleting Page...",
+      },
+      deletePageRequest.bind(this, allWikis, currentPage)
+    );
+  };
+
+  handleViewHistory = () => {
+    this.setState({
+      historyMode: true,
+      viewHistory: true,
+    });
+  };
+
+  oauthCheck = async () => {
+    try {
+      this.setState({
+        spinner: "Connecting to GitHub..."
+      }, async ()=>{
+        const check = (await axios.get(`${BASE_URL}/wikis/oauth-check`)).data
+        if (check.redirect){
+          window.location = check.redirect_url;
+        } else {
+          this.setState({
+            spinner: ""
+          })
+        }
+      })
+    } catch (err) {
+      console.log(err.message);
```

Comment:
Can't we show the error with toast?
Like something went wrong due to which unable to connect ?

Reply:
Done